tests/all-test.php

```diff
@@ -591,4 +591,87 @@ public function test_user_logged_in_multisite() {
 		$output = ob_get_clean();
 		self::assertContains( self::$parsely_html, $output );
 	}
+
+	/**
+	 * Data provider for test_get_current_url
+	 *
+	 * @return array[]
+	 */
+	public function data_for_test_get_current_url() {
+		return array(
+			// Start cases with 'force_https_canonicals' = true
+			array(
+				true,
+				'http://example.com',
+				'https://example.com',
+			),
+			array(
+				true,
+				'https://example.com',
+				'https://example.com',
+			),
+			array(
+				true,
+				'http://example.com:1234',
+				'https://example.com:1234',
+			),
+			array(
+				true,
+				'https://example.com:1234',
+				'https://example.com:1234',
+			),
+			// Start cases with 'force_https_canonicals' = false
+			array(
+				false,
+				'http://example.com',
+				'http://example.com',
+			),
+			array(
+				false,
+				'https://example.com',
+				'http://example.com',
+			),
+			array(
+				false,
+				'http://example.com:1234',
+				'http://example.com:1234',
+			),
+			array(
+				false,
+				'https://example.com:1234',
+				'http://example.com:1234',
+			),
+		);
+	}
+
+	/**
+	 * @category   Function
+	 * @package    SampleTest
+	 * @dataProvider data_for_test_get_current_url
+	 * @covers Parsely::get_current_url
+	 */
+	public function test_get_current_url( $force_https, $url, $expected ) {
+		$options                           = get_option( \Parsely::OPTIONS_KEY );
+		$options['force_https_canonicals'] = $force_https;
+		update_option( \Parsely::OPTIONS_KEY, $options );
+
+		update_option( 'siteurl', $url );
+
+		// Test homepage
+		$this->go_to( '/' );
+		$res = self::$parsely->get_current_url();
+		self::assertSame( strpos( $res, $expected, 0 ), 0 );
+
+		// Test a specific post
+		$post_array = $this->create_test_post_array();
+		$post       = $this->factory->post->create( $post_array );
+		$this->go_to( '/?p=' . $post );
+		$res = self::$parsely->get_current_url();
```

Comment:
You're testing against a post, but the default first argument of the method is `nonpost`. Seems like this should be `post`, along with the `$post` ID (rename variable as $post_id for clarity). Without `post` as the first arg value, there's a whole section of the `get_current_url()` method that isn't being covered in the tests.

Example:

```suggestion
		$post_id    = $this->factory->post->create( $post_array );
		$this->go_to( '/?p=' . $post_id );
		$res = self::$parsely->get_current_url( 'post', $post_id );
```